subset/switches/src/main/java/switchtest/SwitchInterrogator.java

```diff
@@ -396,43 +424,68 @@ private void parseData(String data) {
       }
     } catch (Exception e) {
       System.err.println("Exception parseData:" + e.getMessage());
-      System.exit(0);
     }
   }
 
   private void parseRequestFlag(String data, int requestFlag) {
-    switch (requestFlag) {
-      case 2:
-        // parse show interface
-        login_report += "show interface:\n";
-        parse_packet(data, show_interface_port_expected, show_interface_port_pointers);
-        writeReport();
-        telnetClientSocket.writeData("\n");
-        break;
-      case 3:
-        // parse show platform
-        login_report += "\nshow platform:\n";
-        parse_packet(data, show_platform_port_expected, show_platform_port_pointers);
-        writeReport();
-        telnetClientSocket.writeData("\n");
-        break;
-      case 4:
-        // parse show run
-        data = cleanShowRunData(data);
-        login_report += "\n" + data;
-        writeReport();
-        telnetClientSocket.writeData("\n");
-        break;
-      case 5:
-        // parse show stack
-        login_report += "\nshow stack:\n";
-        parse_inline(data, show_stack_expected, show_stack_pointers, show_stack_data);
-        writeReport();
-        telnetClientSocket.writeData("\n");
-        break;
+    try {
+      switch (requestFlag) {
+        case 2:
+          // parse show interface
+          login_report += "show interface:\n";
+          parse_packet(data, show_interface_port_expected, show_interface_port_pointers);
+          writeReport();
+          telnetClientSocket.writeData("\n");
+          break;
+        case 3:
+          // parse show platform
+          login_report += "\nshow platform:\n";
+          parse_packet(data, show_platform_port_expected, show_platform_port_pointers);
+          writeReport();
+          telnetClientSocket.writeData("\n");
+          break;
+        case 4:
+          // parse show power-inline
+          login_report += "\nshow power-inline:\n";
+          data = trash_line(trash_line(data, 0), 1);
+          parse_inline(data, show_power_expected, show_power_pointers, show_power_data);
+          writeReport();
+          telnetClientSocket.writeData("\n");
+          break;
+        case 5:
+          // parse show run
+          data = cleanShowRunData(data);
+          login_report += "\n" + data;
+          writeReport();
+          telnetClientSocket.writeData("\n");
+          break;
+        case 6:
+          // parse show stack
+          login_report += "\nshow stack:\n";
+          parse_inline(data, show_stack_expected, show_stack_pointers, show_stack_data);
+          writeReport();
+          telnetClientSocket.writeData("\n");
+          break;
+      }
+    } catch (Exception e) {
+      System.err.println("Exception parseRequestFlag:" + e.getMessage());
+      System.exit(0);
```

Comment:
0 means success -- how is an exception success?  Error return codes should be non-zero

Reply:
fixed System.exit(1)